Tarteeb.importer/Services/Clients/ClientServices.cs

```diff
@@ -14,25 +18,65 @@ internal class ClientServices
     {
         public readonly StorageBroker storageBroker;
         public readonly LoggingBroker loggingBroker;
-        public ClientServices(StorageBroker storageBroker,LoggingBroker loggingBroker)
+        public ClientServices(StorageBroker storageBroker, LoggingBroker loggingBroker)
         {
             this.storageBroker = storageBroker;
             this.loggingBroker = loggingBroker;
         }
 
+        private dynamic isInvalid(Guid id) => new 
+        {
+            Condition = id == default,
+            Message = "Id is required"
+        };
+
+        private dynamic isInvalid(string text) => new 
+        {
+            Condition = string.IsNullOrWhiteSpace(text),
+            Message = "Text is required"
+        };
+
+        private dynamic isInvalidEmail(string email) => new
+        {
+            Condition = !Regex.IsMatch(email, "^[\\w-\\.]+@([\\w-]+\\.)+[\\w-]{2,4}$"),
+            Message = "Text is required"
+        };
+
         public async Task<Client> AddClientAsync(Client client)
         {
             try
             {
+                Validate(
+                    (isInvalid(client.Id), nameof(client.Id)),
+                    (isInvalid(client.Firstname), nameof(client.Firstname)),
+                    (isInvalidEmail(client.Email), nameof(client.Email))
+                    );
+
                return await storageBroker.InsertClientAsync(client);
             }
-            catch (Exception)
+            catch (ArgumentNullException)
```

Comment:
1. Return back the if null check. 
2. Instead of catching ArgumentNullException, now catch NullClientException
3.